client/src/components/Viewer/Viewer.js

```diff
@@ -55,6 +55,9 @@ function Viewer(props) {
         setIsConnected(true);
         socket.emit('addUser', {
           token: JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
+          user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user'][
+            'user_email'
+          ],
```

Comment:
The variable name `user_id` may be misleading since it actually holds the `user_email`. Consider renaming it to `user_email` for better clarity.
```diff
-           user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user'][
+           user_email: JSON.parse(localStorage.getItem('dfs-user'))?.['user']?.['user_email'],
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*